src/query/graphite/native/builtin_functions.go

```diff
@@ -94,6 +94,45 @@ func sortByMaxima(ctx *common.Context, series singlePathSpec) (ts.SeriesList, er
 	return highestMax(ctx, series, len(series.Values))
 }
 
+/*
+useSeriesAbove compares the maximum of each series against the given `value`. If the series
+maximum is greater than `value`, the regular expression search and replace is
+applied against the series name to plot a related metric
+
+e.g. given useSeriesAbove(ganglia.metric1.reqs,10,'reqs','time'),
+the response time metric will be plotted only when the maximum value of the
+corresponding request/s metric is > 10
+
+&target=useSeriesAbove(ganglia.metric1.reqs,10,"reqs","time")
+*/
+func useSeriesAbove(ctx *common.Context, seriesList singlePathSpec, maxAllowedValue float64, search, replace string) (ts.SeriesList, error) {
+	var newNames []string
+
+	for _, series := range seriesList.Values {
+		if series.SafeMax() > maxAllowedValue {
+			seriesName := strings.Replace(series.Name(), search, replace, -1)
+			newNames = append(newNames, seriesName)
+		}
+	}
+	allRenamedSeries := strings.Join(newNames, ",")
+	if len(allRenamedSeries) > 0 {
+		target := strings.ReplaceAll("group(%)", "%", allRenamedSeries)
+		eng := NewEngine(ctx.Engine.Storage())
+		expression, err := eng.Compile(target)
+		if err != nil {
+			return ts.NewSeriesList(), err
+		}
+		resultSeriesList, err := expression.Execute(ctx)
```

Comment:
@arnikola do you think we need to add concurrency here in the same way we added it to #2654 ?

Reply:
Would probably be a good idea here too, yeah